### DISCUSSION TITLE
Question with SHARPy example 'Flutter Analysis of a Goland Wing using the SHARPy Linear Solver' - Inputs to the linear system

### DISCUSSION THREAD

**Discussion by Alanpwu** [ImperialCollegeLondon/sharpy#134] [Q&A] [answered]
In the example of 'Flutter Analysis of a Goland Wing using the SHARPy Linear Solver', the number of input of the linearised aeroelastic system is always (No. of structural states)1.5 + 2(No. of control surface).  ### I am wandering what these structural inputs represent?

**Comment by ngoiz** [comment, marked as answer]
The inputs to the aeroelastic system are the aerodynamic inputs followed by the structural inputs. The aero inputs are modal displacements and modal velocities in this case. The structural inputs are modal forces. Hence, the number of inputs is 3 times the number of modes plus any control surfaces/gusts etc. It should be fairly straight forward from the docs https://ic-sharpy.readthedocs.io/en/master/includes/linear/assembler/linearaeroelastic/LinearAeroelastic.html



You can also have a look at this paper https://arc.aiaa.org/doi/abs/10.2514/6.2021-1798 for a more in-depth review of how the aeroelastic system is assembled.
